pkg/querier/handler.go

```diff
@@ -0,0 +1,102 @@
+package querier
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/grafana/loki/pkg/loghttp"
+	"github.com/grafana/loki/pkg/logproto"
+	"github.com/grafana/loki/pkg/querier/queryrange"
+	"github.com/grafana/loki/pkg/querier/queryrange/queryrangebase"
+)
+
+type Handler struct {
+	api *QuerierAPI
+}
+
+func NewQuerierHandler(api *QuerierAPI) *Handler {
+	return &Handler{
+		api: api,
+	}
+}
+
+func (h *Handler) Do(ctx context.Context, req queryrangebase.Request) (queryrangebase.Response, error) {
+	switch concrete := req.(type) {
+	case *queryrange.LokiRequest:
+		res, err := h.api.RangeQueryHandler(ctx, concrete)
+		if err != nil {
+			return nil, err
+		}
+
+		params, err := queryrange.ParamsFromRequest(req)
+		if err != nil {
+			return nil, err
+		}
+
+		return queryrange.ResultToResponse(res, params)
+	case *queryrange.LokiInstantRequest:
+		res, err := h.api.InstantQueryHandler(ctx, concrete)
+		if err != nil {
+			return nil, err
+		}
+
+		params, err := queryrange.ParamsFromRequest(req)
+		if err != nil {
+			return nil, err
+		}
+
+		return queryrange.ResultToResponse(res, params)
+	case *queryrange.LokiSeriesRequest:
+		request := &logproto.SeriesRequest{}
```

Comment:
Probably a bug.

Reply:
yeah, my guess would be we can just pass `req` to `h.api.SeriesHnadler` on the next line

Reply:
I've fixed it.